query-graphs/src/tree-rendering.ts

```diff
@@ -58,22 +58,22 @@ const orientations: {[k in GraphOrientation]: Orientation} = {
         x: (d, viewSize) => viewSize.y - d.y,
         y: (d, _viewSize) => d.x,
         textdimension: "x",
-        textdimensionoffset: d => (d.children || d._children ? 10 : -10),
-        textanchor: d => (d.children || d._children ? "start" : "end"),
+        textdimensionoffset: d => (d.children ? 10 : -10),
+        textanchor: d => (d.children ? "start" : "end"),
         nodesize: maxLabelLength =>
             [11.2 /* table node diameter */ + 2, maxLabelLength * 6 + 10 /* textdimensionoffset */] as d3point,
```

Comment:
Could have a minimum here as well? Queries without the Plan and Actual row counts on the labels are fine, but with the row counts there's overlap. It's far from perfect anyway. I suppose the counts might better moved from the middle of the edge. Modified postgres/limitscan.json:

[
  {
    "Plan": {
      "Node Type": "Limit",
      "Parallel Aware": false,
      "Startup Cost": 0.00,
      "Total Cost": 0.01,
      "Plan Rows": 1,
      "Plan Width": 8,
      "Actual Startup Time": 0.001,
      "Actual Total Time": 0.001,
      "Actual Rows": 0,
      "Actual Loops": 1,
      "Output": ["r1", "r2"],
      "Plans": [
        {
          "Node Type": "Result",
          "Parent Relationship": "Outer",
          "Parallel Aware": false,
          "Startup Cost": 0.00,
          "Total Cost": 0.01,
          "Plan Rows": 1,
          "Plan Width": 8,
          "Actual Startup Time": 0.000,
          "Actual Total Time": 0.000,
          "Actual Rows": 0,
          "Actual Loops": 0,
          "Output": ["r1", "r2"]
        }
      ]
    },
    "Planning Time": 0.028,
    "Triggers": [
    ],
    "Execution Time": 0.013
  }
]

Reply:
Added a minimum space of 90 pixel.

That solves the issue for your example and seems like a good idea in general.
However, it does not completely solve the issues, e.g. if instead of "Limit" the node label was a longer string. I don't know of a generally nice solution for left-to-right/right-to-left. I guess one possibility would be to move the labels below the nodes, also in the horizontal layouts... Something we might want to do as part of the re-design for #56 anyway...